cluster/permissions.go

```diff
@@ -389,19 +395,19 @@ func buildPermissionsForRows(ctx *Context, rows *sql.Rows) ([]*Permission, error
 		return nil, err
 	}
 	// get the actions
-	err = getActionsForPermissions(ctx, permissionsHash)
+	err = getActionsForPermissionsInTx(ctx, permissionsHash, inTx)
 	if err != nil {
 		return nil, err
 	}
 	// get the resources
-	err = getResourcesForPermissions(ctx, permissionsHash)
+	err = getResourcesForPermissionsInTx(ctx, permissionsHash, inTx)
 	if err != nil {
 		return nil, err
 	}
 	return permissions, nil
 }
 
-func getActionsForPermissions(ctx *Context, permsMap map[uuid.UUID]*Permission) error {
+func getActionsForPermissionsInTx(ctx *Context, permsMap map[uuid.UUID]*Permission, inTx bool) error {
```

Comment:
what is the main reason behind `inTx` ?  When should we use it? is it really needed?

Reply:
Some times we need to do queries in a transation and sometimes outside of it